torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -34,37 +30,25 @@ ncclComm_t NCCLComm::getNcclComm() {
             ". ",
             commFailureMsg));
   }
-  // only wait for initialization if nonblocking mode is enabled
-  if (!initialized_ && nccl_use_nonblocking()) {
-    waitUntilInitialized(nccl_nonblocking_timeout());
+  if (!initialized_) {
```

Comment:
Previous implementation should not hang forever? as there is time bound on timeout?

Reply:
It will not hang forever, if it is just regular hang.
But it will hang forever, if there is an async error using init.

Reply:
If non-blocking init itself failed, does NCCL know (we are in non-blocking mode?) to populate the error in `ncclGetAsyncError` ? (I'm assuming `ncclGetAsyncError` only works in nonblocking mode.

Reply:
`ncclGetAsyncError` will populate error when non-blocking init fails. 

`ncclGetAsyncError` also works in blocking mode, i.e. when checking if a running collective hit any network error.